src/core/internal/traits.d

```diff
@@ -258,14 +258,63 @@ template hasElaborateCopyConstructor(S)
     }
     else static if (is(S == struct))
     {
-        enum hasElaborateCopyConstructor = __traits(hasMember, S, "__xpostblit");
+        enum hasElaborateCopyConstructor = __traits(hasMember, S, "__xpostblit") ||
+             () {
+                static if (__traits(hasMember, S, "__ctor"))
+                {
+                    static foreach (f; __traits(getOverloads, S, "__ctor"))
+                    {{
+                        bool r = __traits(compiles, {
+                                auto p = &f;
+
+                                // Check if ctor is callable with lval or rval
+                                S s;
```

Comment:
This won't work with function-local structs and probably with @disabled structs. `S s = S.init` should fix your static assert issues.

Reply:
Thanks for the help.

Now I'm also taking into account `@disabled` structs.